hphp/runtime/ext/datetime/ext_datetime.php

```diff
@@ -876,8 +875,21 @@ function timezone_name_get(DateTimeZone $timezone): string {
 }
 
 <<__ParamCoerceModeFalse>>
-function timezone_open(string $timezone): DateTimeZone {
-  return new DateTimeZone($timezone);
+function timezone_open(string $timezone): mixed {
+  try {
+    return new DateTimeZone($timezone);
+  }
+  catch (Exception $e) {
+    trigger_error(
+      str_replace(
+        'DateTimeZone::__construct():',
+        __FUNCTION__ . '():',
+        $e->getMessage()
+      ),
+      E_WARNING
```

Comment:
Sounds logical, otherwise I'd have a real pain in the ass to deal with.
Does E_STRICT have no use cases?